src/librustc/mir/repr.rs

```diff
@@ -699,6 +733,9 @@ pub enum StatementKind<'tcx> {
 
     /// End the current live range for the storage of the local.
     StorageDead(Lvalue<'tcx>),
+
+    /// No-op. Useful for deleting instructions without affecting statement indices.
+    Nop,
```

Comment:
Potential point of disagreement (we already have at least one lazy patching mechanism which preserves indices).